### ISSUE TITLE
Not Supporting laravel/framework v7.18.0

### ISSUE THREAD

**Issue by dhruva81** [ZsgsDesign/laravel-pdf-latex#7]
Hi 
This package is not supporting laravel/framework v7.18.0.

**Comment by ZsgsDesign** [comment]
Could you kindly share with us your error log regarding this issue?

**Comment by dhruva81** [comment]
Thank you for your reply.
Please find the attached image.
![image](https://user-images.githubusercontent.com/42173738/87328329-f82e5e80-c552-11ea-8cfb-a7df197b027e.png)
![image](https://user-images.githubusercontent.com/42173738/87328386-067c7a80-c553-11ea-9885-79b5d8f6e487.png)

**Comment by ismaelw** [comment]
Having the same issue. Would like to use this Package with Laravel 7 but can't because of the error mentioned above.

**Comment by ismaelw** [comment]
@dhruva81 did you find a solution for this issue?

**Comment by ZsgsDesign** [comment]
```json
    "require": {
        "php": ">=5.4.0",
        "illuminate/support": "5.4.x|5.5.X|5.6.X|5.7.X|5.8.X|^6.0",
        "illuminate/filesystem": "5.4.x|5.5.X|5.6.X|5.7.X|5.8.X|^6.0",
        "symfony/process": "v2.x|v3.x|v4.x"
    },
```
just add `^7.0` here.

I am having an oral examination these days, would fix this later. Sending a PR would be fine.

**Comment by GeorgeGhiottone** [comment]
Pull request done!  #8

**Comment by ZsgsDesign** [comment]
Just released `2.1.6`, just run `composer update`.

If this got solved, feel free to close this issue.

**Comment by dhruva81** [comment]
![image](https://user-images.githubusercontent.com/42173738/92201302-d047e100-ee99-11ea-9737-d72517e2ba2c.png)
![image](https://user-images.githubusercontent.com/42173738/92201327-e190ed80-ee99-11ea-9b79-23877b1e477a.png)

**Comment by ismaelw** [comment]
Having the same issue here with laravel 7.27.0.
Maybe "symfony/process" also needs a higher version?

**Comment by ZsgsDesign** [comment]
@ismaelw Yes, OMG I forgot about that.

I am still using Laravel 5.8 LTS, looks like `7.x` made a lot of changes lol.

**Comment by ZsgsDesign** [comment]
@ismaelw Is Laravel 7 now using Symfony 5? If so you can PR directly or just reply here, I will make that correction once I read the message.

**Comment by ismaelw** [comment]
@ZsgsDesign I'm somehow not able to make a PR.
But I can confirm that Laravel 7 is now using Symfony 5.

Also I wanted to ask you about the part where you check for Linux and throw a Unspported Operating System. Is this necessary? I remember your package failing there but after removing it from the code it worked.

Thank you so much!

**Comment by ZsgsDesign** [comment]
> @ZsgsDesign I'm somehow not able to make a PR.
> But I can confirm that Laravel 7 is now using Symfony 5.
> 
> Also I wanted to ask you about the part where you check for Linux and throw a Unspported Operating System. Is this necessary? I remember your package failing there but after removing it from the code it worked.
> 
> Thank you so much!

This package is built for Linux only, so we have it checked.

```php
if (!in_array(PHP_OS_FAMILY, ['Linux'])){
    throw new LatextException("Unsupported Operating System");
}
```

Are you using Windows or Mac and it succeeded? Give me feedback.

**Comment by ismaelw** [comment]
I'm on an external hoster. The server running there with pdflatex is running a Linux System.
Still this package failed. That's why I thought it was weird. Anyway because of the external hoster I have to set the specific binPath to pdflatex so it works. But still this part of the PHP check I had to remove.

**Comment by ZsgsDesign** [comment]
> I'm on an external hoster. The server running there with pdflatex is running a Linux System.
> Still this package failed. That's why I thought it was weird. Anyway because of the external hoster I have to set the specific binPath to pdflatex so it works. But still this part of the PHP check I had to remove.

How about run `PHP_OS_FAMILY` yourself and see the result?

```php
<?PHP
    var_dump(PHP_OS_FAMILY);
```

Maybe it's a string? I am also confused.

**Comment by ismaelw** [comment]
> > I'm on an external hoster. The server running there with pdflatex is running a Linux System.
> > Still this package failed. That's why I thought it was weird. Anyway because of the external hoster I have to set the specific binPath to pdflatex so it works. But still this part of the PHP check I had to remove.
> 
> How about run `PHP_OS_FAMILY` yourself and see the result?
> 
> ```
> <?PHP
>     var_dump(PHP_OS_FAMILY);
> ```
> 
> Maybe it's a string? I am also confused.

With the following test i receive the also following output:

`<?PHP
	echo PHP_OS .  ' ---- ';
	var_dump(PHP_OS_FAMILY);
?>`

Result:

FreeBSD --- string(3) "BSD"

PHP_OS_FAMILY is available from PHP version 7.2. On my hosting I am using PHP 7.4.
As said in the PHP Documentation PHP_OS_FAMILY outputs a string of either 'Windows', 'BSD', 'Darwin', 'Solaris', 'Linux' or 'Unknown'. Maybe you could fix the code using Linux, Solaris, Darwin and BSD instead of just Linux :)
